### PR TITLE
DOC: move third-party packages to new page

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

We now have a lighter third-party page at https://matplotlib.org/mpl-third-party that we hope will have a lower bar to entry than the current page (https://matplotlib.org/stable/thirdpartypackages/index.html).  This PR moves the page.  

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by jklymak** [comment]
https://57108-1385122-gh.circle-artifacts.com/0/doc/build/html/index.html

When (if) this goes in, and the site gets "released" I'd suggest http://matplotlib.org/mpl-third-party will need to go to a one external review standard that we usually use for documentation.

**Review by yezz123 — APPROVED** [review]

File: doc/devel/documenting_mpl.rst [hunk 1]
```diff
@@ -290,12 +290,12 @@ Including figures and files
 ---------------------------
 
 Image files can directly included in pages with the ``image::`` directive.
-e.g., :file:`thirdpartypackages/index.rst` displays the images for the third-party
-packages as static images::
+e.g., :file:`tutorials/intermediate/constrainedlayout_guide.py` displays 
+a couple of images as static images::
```
Comment (story645) [inline doc/devel/documenting_mpl.rst:294]:
```suggestion
e.g., :file:`tutorials/intermediate/constrainedlayout_guide.py`  displays examples of the feature as static images::
```
images as images sounds weirds

File: doc/thirdpartypackages/index.rst [hunk 2]
```diff
@@ -1,4 +1,10 @@
-.. _thirdparty-index:
+:orphan:
+
+.. note:: 
+
+    This page has been moved to <https://matplotlib.org/mpl-third-party> 
+    where you will find an up-to-date list of packages.
```
Comment (story645) [inline doc/thirdpartypackages/index.rst:6]:
```suggestion
    This page has been moved to <https://matplotlib.org/mpl-third-party>, 
    where you will find an up-to-date list of packages.  
```

File: doc/devel/documenting_mpl.rst (hunk 1 above)
Comment (jklymak) [inline doc/devel/documenting_mpl.rst:294]:
```suggestion
e.g., :file:`tutorials/intermediate/constrainedlayout_guide.py` displays 
a couple for static images::
```

**Review by story645 — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

**Pull request merged**